File: client/utilities/activityManager.py
import json

from clients.httpSingleton import HTTPSingleton
from models.activity import Activity


class ActivityManager:
    __instance__ = None

    def __init__(self, e_id):
        ActivityManager.__instance__ = Activity(e_id)

    @staticmethod
    def create_activity(e_id):
        ActivityManager(e_id)

    @staticmethod
    def add_set(set):
        ActivityManager.__instance__.sets.append(set)

    @staticmethod
    def get_activity():
        return ActivityManager.__instance__

    @staticmethod
    def post_activity():
        HTTPSingleton.__activity_client__.save_activity(ActivityManager.__instance__.to_json_format())
        ActivityManager.__instance__ = None

    @staticmethod
    def reset():
        ActivityManager.__instance__ = None

File: client/Bachelor-Client/Bachelor-Client/Utils/ActivityManager.cs
using Bachelor_Client.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bachelor_Client.Utils
{
    public class ActivityManager
    {
        private static Activity _instance;

        public static void CreateActivity(string e_id)
        {
            ActivityManager._instance = new Activity { e_id = e_id, date = new DateTime(), sets = new List<Set>() };
        }

        public static void AddSet(Set set)
        {
            ActivityManager._instance.sets.Add(set);
        }

        public static Activity GetActivity()
        {
            return ActivityManager._instance;
        }

        public static void Reset()
        {
            ActivityManager._instance = new Activity {};
        }
    }
}
